src/main.rs

```diff
@@ -128,3 +130,19 @@ fn find_zone(identifier: &Name) -> anyhow::Result<Name> {
         _ => Result::Err(anyhow::format_err!("unhandled reponse type")),
     }
 }
+
+#[instrument]
+pub fn search(identifier: Name, rtype: RecordType) -> anyhow::Result<DnsResponse> {
+    let client = SyncClient::new(
+        UdpClientConnection::new(
+            *RESOLVER
+                .get()
+                .context("could not read resolver from config")?,
+        )
+        .context("failed to connect to recursor")?,
+    );
+
+    let query = || Ok(client.query(&identifier, DNSClass::IN, rtype)?);
+
+    Ok(retry(ExponentialBackoff::default(), query)?)
```

Comment:
uh not really, this just gets called most frequently, so I implemented it here first.
should I do it for the updates aswell?

if yes I should also map the errors properly to only retry for errors where retries might help, i.e not auth errors so it will be more work

Reply:
I'd be inclined to say yes.